active/AnimatedIcon/AnimatedIconSpec.md

```diff
@@ -0,0 +1,894 @@
+{
+    AnimatedSettingsVisualSource();
+};
+
+[webhosthidden]
+runtimeclass AnimatedFindVisualSource
+    : [default] Microsoft.UI.Xaml.Controls.IAnimatedVisualSource, Microsoft.UI.Xaml.Controls.IAnimatedVisualSource2
+{
+    AnimatedFindVisualSource();
+};
+
+[webhosthidden]
+runtimeclass AnimatedChevronDownSmallVisualSource
+    : [default] Microsoft.UI.Xaml.Controls.IAnimatedVisualSource, Microsoft.UI.Xaml.Controls.IAnimatedVisualSource2
+{
+    AnimatedChevronDownSmallVisualSource();
+};
+
+[webhosthidden]
+runtimeclass AnimatedChevronUpDownSmallVisualSource
+    : [default] Microsoft.UI.Xaml.Controls.IAnimatedVisualSource, Microsoft.UI.Xaml.Controls.IAnimatedVisualSource2
+{
+    AnimatedChevronUpDownSmallVisualSource();
+};
+
+[webhosthidden]
+runtimeclass AnimatedChevronRightDownSmallVisualSource
+    : [default] Microsoft.UI.Xaml.Controls.IAnimatedVisualSource, Microsoft.UI.Xaml.Controls.IAnimatedVisualSource2
+{
+    AnimatedChevronRightDownSmallVisualSource();
+};
+
+```
+
+# Appendix
+
+## Accessibility
+
+This control is going to be part of the Icon family of controls so the accessibility requirements will reflect what the other icon controls have. AnimatedIcon will need to ensure that the high contrast brushes propagate correctly if the user changes their settings to a high contrast theme.  
+
+If the user turns off animations in theri system settings, the AnimatedIcons will not animate. 
+
+## Adding an instance state property 
+
+In a future version of this control, we would like to add an instanced state property to the control so Intellisense can inform the developer of the State property. Today, the parser cannot recognize a dependency property and does not have the ability to tell the difference between a dependency property and an instance property of the same name. There would need to be a fix to the parser in order for this to work. 
+
+### Updates to IconSource to support AnimatedIconSource
+
+Since some controls take an IconElement and others take an IconSource, we are going to add two methods (public CreateIconElement and 
+protected CreateIconElementCore) to IconSource. This will help cover the two sccenarios below: 
+
+1.	I’m a developer creating a new control and I want to expose an IconSource property where consumers of my control can set ‘any’ iconsource. This control developer should not have to worry about each type of IconSource like FontIconSource, BitmapIconSource, AnimatedIconSource etc. In this case the developer can just use IconSource and call IconSource.CreateIconElement to get the corresponding IconElement without knowing what specific IconSource/Element it is. 
+2.	I’m a developer creating a new type of IconSource, say Animated2IconSource. I would have to override Animated2IconSource.CreateElementCore and return an Animated2IconElement so that my new IconSource will work anywhere an IconSource can be used.
```

Comment:
What about

3. I'm a developer who already created a new type of IconSource, say, AppxPackageIconSource and shipped it for use with mux:v2. When the app updates to the new version of mux, my AppxPackageIconSource.CreateIconElement method will not work, because I did not implement CreateIconElementCore.

Reply:
I think this is equivalent to: I wrote an IconSource subclass but didn't do the override (we don't have the ability to make this an abstract method). I think we should just have a documented error.